### ISSUE TITLE
Replicate outlines vLLM performance improvements

### ISSUE THREAD

**Issue by laurens-gs** [noamgat/lm-format-enforcer#125]
On a system with a 1xH100 80GB GPU, vLLM and LLaMa 70B (FP8) I go from 1000+ tok/s to around 100 tok/s when I enable lm-format-enforcer. Profiling revealed that the core parsing by lm-format-enforcer is sufficiently fast. The large slowdown seems to come from applying the logit mask.

Outlines vLLM are changing how the logits are passed around, see also the progress here: https://github.com/vllm-project/vllm/issues/3567

I would like to request a similar strategy is replicated for lm format enforcer, otherwise the library is rendered unusable for large batch processes.
